pkg/controllers/dashboard/kubernetesprovider/kubernetesprovider.go

```diff
@@ -33,10 +33,15 @@ func Register(ctx context.Context,
 
 func (h *handler) OnChange(key string, cluster *v3.Cluster) (*v3.Cluster, error) {
 	if cluster == nil || cluster.Status.Provider != "" {
+		// rke1 windows clusters check
 		// if cluster has windows enabled and provider is not rke.windows, continue to detect if it has windows nodes
 		if cluster == nil || !cluster.Spec.WindowsPreferedCluster || cluster.Status.Provider == "rke.windows" {
 			return cluster, nil
 		}
+		// rke2 windows clusters check
+		if cluster == nil || cluster.Status.Provider == "rke2.windows" {
+			return cluster, nil
+		}
```

Comment:
Per offline: merge the checks and swap to a endswith check since harvester will probably be coming soon